### PR TITLE
Saving target calories

### PR DESCRIPTION
**Pull request by ahz0001**
Added calculations for exercise level and weight goals as well as tests for target calorie calculations.

### CODE REVIEW

**Comment by omidafshar** [comment]
Looks like there's a merge conflict with the master branch for the Users controller. The code can't be merged until this is resolved.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -6,21 +6,46 @@ class UsersController < ApplicationController
...
+	    @weight = :weight
+	    @height = :height
+	    @exercise = :exercise
+	    @goal = :goal
+	    @budget = :budget
+	    @time = :time
+	    @cuisine = :cuisine
+	    @calories = calc_calories(@gender, @weight, @height, @age, @exercise, @goal)
+	end
+
```
Comment (cheolhojang) [inline app/controllers/users_controller.rb:19]: good idea to short variables. i think it def helps us in the long run. also should we cite where we found the calculation somewhere in the comments?

**Review by nkompella — COMMENTED** [review]
Where did you get the formula to calculate calories from?

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ahz0001) [reply app/controllers/users_controller.rb:19]: Yeah, I added links to the places I found those calculations and numbers

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -14,13 +14,41 @@ def show
...
+		end
+		#If you are sedentary and do not exercise, multiply your 
+		#BMR by 1.2. If you exercise lightly one to three times 
+		#per week, multiply by 1.375. If you exercise three to 
+		#five days per week, multiply by 1.55. For exercise six 
+		#or seven days per week, multiply by 1.725; if you exercise 
+		#seven days a week and also have a physically demanding job, 
+		#multiply by 1.9.
+		#https://www.livestrong.com/article/526442-the-activity-factor-for-calculating-calories-burned/
+
```
Comment (cheolhojang) [inline app/controllers/users_controller.rb:45]: Good looks providing the source of the calculation formula

**Comment by nkompella** [comment]
I like that you modularized the controller by creating a separate function for calculating calories, and that you resolved my earlier comment by including the source for the formula! All the tests seem comprehensive as well.

**Pull request merged**